### PR TITLE
[Core][Nurbs-Geometries] Refactor Curve Integration

### PR DESCRIPTION
**Pull request by tteschemacher**
**Description**
This PR only refactors the creation of integration points from internal functions to a single utility function. This helps maintenance and additionally may be used by various 1D-based geometries in the future.

### CODE REVIEW

File: kratos/geometries/brep_curve_on_surface.h [hunk 1]
```diff
@@ -334,7 +334,7 @@ class BrepCurveOnSurface
      */
     void Spans(std::vector<double>& rSpans, IndexType DirectionIndex = 0) const
```
Comment (dagmawibekel) [inline kratos/geometries/brep_curve_on_surface.h:335]: Shouldn't you get rid of the DirectionIndex in the argument list

File: kratos/integration/integration_point_utilities.cpp [hunk 2]
```diff
@@ -15,6 +15,33 @@
...
+    /* @brief Creates integration points within provided spans according
+     *        to the provided points per span.
+     * @param rIntegrationPoints result integration points.
+     * @param rSpansLocalSpace spans in which integration points are generated.
+     * @param IntegrationPointsPerSpan number of integration points per span.
+     */
+    void IntegrationPointUtilities::CreateIntegrationPoints1D(
+        IntegrationPointsArrayType& rIntegrationPoints,
+        const std::vector<double>& rSpansLocalSpace,
+        SizeType IntegrationPointsPerSpan)
```
Comment (dagmawibekel) [inline kratos/integration/integration_point_utilities.cpp]: can't `IntegrationPointsPerSpan` be const?

File: kratos/geometries/brep_curve_on_surface.h (hunk 1 above)
Comment (tteschemacher) [reply kratos/geometries/brep_curve_on_surface.h:335]: True, however, the specific interface will be refactored in #8490 . ;) Sorry for the many PRs.

File: kratos/integration/integration_point_utilities.cpp (hunk 2 above)
Comment (tteschemacher) [reply kratos/integration/integration_point_utilities.cpp]: Hmm good point. I quite remember that it wasn't more efficient.

@philbucher integer and size types were ideally passed without reference nor const, right?

**Comment by tteschemacher** [comment]
Thanks a lot @dagmawibekel for the review 👍

File: kratos/integration/integration_point_utilities.cpp (hunk 2 above)
Comment (philbucher) [reply kratos/integration/integration_point_utilities.cpp]: afaik (@roigcarlo please correct me if I am wrong)
`int`: 32bit
`std::size_t`: 64bit
`int&`: 64bit
`std::size_t&`: 64bit

const or not is up to you => what do you intend? Should this value be mutable? If no, then make it const

Comment (tteschemacher) [reply kratos/integration/integration_point_utilities.cpp]: Thanks for the explanation. 👍 

It's not touched do it'd be better to make it const then.

Comment (tteschemacher) [reply kratos/integration/integration_point_utilities.cpp]: sweet. it is const now.

**Comment by tteschemacher** [comment]
Hey guys, can we merge that stuff?

**Review by philbucher — APPROVED** [review]

**Comment by tteschemacher** [comment]
:H

**Pull request merged**